### PR TITLE
feat: add task 4

### PR DESCRIPTION
**Pull request by stsepchanka**

### CODE REVIEW

File: src/data-access/script.sql [hunk 1]
```diff
@@ -6,3 +6,13 @@ insert into users (login, password, age) values ('petr_petrov', 'petrov1', 30);
 insert into users (login, password, age) values ('ivan_ivanov', 'ivanov1', 25);
 
 SELECT * FROM users;
+
+
+create table groups (id UUID PRIMARY KEY DEFAULT uuid_generate_v4(), name text, permissions text);
```
Comment (kuzanatoliy) [inline src/data-access/script.sql]: Would be nice if you used sequelize-cli and created migrations

File: src/models/group.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import * as Joi from "joi";
+import { DataTypes, Model, UUIDV4 } from "sequelize";
+import { sequelize } from "../data-access/connectDB";
+
+export const PERMISSIONS = ["READ", "WRITE", "DELETE", "SHARE", "UPLOAD_FILES"];
```
Comment (kuzanatoliy) [inline src/models/group.ts]: I would use enum here

File: src/routers/groups.router.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    .get(groupController.getGroupById())
+    .put(validateSchema(groupSchema), groupController.updateGroup())
+    .delete(groupController.deleteGroup());
+
+  router
+    .route("/")
+    .get(groupController.getGroups())
+    .post(validateSchema(groupSchema), groupController.addGroup());
+
+  router.route("/:id/add-users").post(groupController.addUsersToGroup());
```
Comment (kuzanatoliy) [inline src/routers/groups.router.ts]: I think "/:id/users" would be enough here

File: src/services/group.service.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    return group.toDomain();
+  }
+
+  async getGroupByName(groupName: string): Promise<IGroup> {
+    const group = await Group.findOne({
+      where: { name: groupName },
+    });
+
+    if (!group) {
```
Comment (kuzanatoliy) [inline src/services/group.service.ts]: I wouldn't use errors for handling empty results

File: src/routers/group.controller.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      }
+    };
+  }
+
+  deleteGroup() {
+    return async (req: Request, res: Response): Promise<void> => {
+      try {
+        const groupId = req.params.id;
+        await this.groupService.deleteGroupById(groupId);
+        res.status(StatusCodes.NO_CONTENT).end();
```
Comment (kuzanatoliy) [inline src/routers/group.controller.ts]: Just use .sendStatus

**Comment by kuzanatoliy** [comment]
config.ts - would be nice to add default values. And I don't see reasons to use config object

File: src/data-access/script.sql (hunk 1 above)
Comment (stsepchanka) [reply src/data-access/script.sql]: Sequelize-cli package has been added and used.

File: src/routers/groups.router.ts (hunk 3 above)
Comment (stsepchanka) [reply src/routers/groups.router.ts]: Changed. Thanks.

File: src/models/group.ts (hunk 2 above)
Comment (stsepchanka) [reply src/models/group.ts]: Permission enum has been added.

File: src/routers/group.controller.ts (hunk 5 above)
Comment (stsepchanka) [reply src/routers/group.controller.ts]: Changed. Thanks

File: src/services/group.service.ts (hunk 4 above)
Comment (stsepchanka) [reply src/services/group.service.ts]: Changed. Thanks.

**Pull request merged**